### ISSUE TITLE
Add left, right and bottom location options for TabContainer

### ISSUE THREAD

**Issue by EricEzaM** [godotengine/godot-proposals#1986]
### Describe the problem or limitation you are having in your project
More advanced usage of tabs.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Add left, right and bottom options for TabContainer

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
**Bottom: Visual Studio**
![image](https://user-images.githubusercontent.com/41730826/102222894-c1a8a580-3f2f-11eb-90a5-8653e3558f35.png)

**Left/Right**
![image](https://user-images.githubusercontent.com/41730826/102222967-d71dcf80-3f2f-11eb-8a81-dc1ed7981c98.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?
Basically would need to write your own tab container from scratch

**Comment by Calinou** [comment]
I'm not sure if side tab locations are that useful to have, since this can be replicated relatively easily using themed Button nodes.

**Comment by Shadowblight** [comment]
> See also [godotengine/godot#4214](https://github.com/godotengine/godot/issues/4214).
> 
> Tabs on the bottom sound good, but I'm not sure if side tab locations are that useful to have in TabContainer, since this can be replicated relatively easily using themed Button nodes.

Hi, perhaps I'm a bit too ignorant, but I am requesting further clarification on this.

Aren't TabContainers on the whole just pre-packaged "themed buttons" to begin with in their current implementation on top/bottom? They're just pre-coded so we don't have to worry about doing them manually, which is largely what a game engine is supposed to do: provide ease of use for common needs in creating games. If I want left or right aligned tabs but with themed buttons as this thread suggests, it'd involve manually coding the whole concept of what the TabContainer does.

Why is it considered merited as worth it to implement TabContainers with tabs on top/bottom, but not left/right?

**Comment by Calinou** [comment]
> They're just pre-coded so we don't have to worry about doing them manually, which is largely what a game engine is supposed to do: provide ease of use for common needs in creating games.

As mentioned in the Best practices for engine contributors, [the problem has to be complex or frequent](https://docs.godotengine.org/en/stable/contributing/development/best_practices_for_engine_contributors.html#the-problem-has-to-be-complex-or-frequent) to be worth including in core.

Remember that adding support for horizontal tab layouts is a *lot* more work than just allowing tabs to be placed at the top or bottom, as there is little code in common when doing this. New StyleBoxes also need to be designed for the default project theme and editor theme for horizontal tabs, which further adds to the complexity.
